### PR TITLE
Use EnvironmentContributingAction to toggle auto-injection

### PR DESCRIPTION
**Pull request by welandaz**
This change switches from using `EnvironmentContributor` to control the Gradle Plugin/Maven Extension injection to `EnvironmentContributingAction` instead, as this happens to support both Freestyle and Pipeline job configurations

### CODE REVIEW

File: src/main/java/hudson/plugins/gradle/injection/GitScmListener.java [hunk 1]
```diff
@@ -116,26 +122,37 @@ private static boolean shouldDisableMavenInjection(InjectionConfig config) {
     }
 
     /**
-     * Action that holds Maven environment variables to be set in {@link MavenInjectionEnvironmentContributor}.
+     * Action that disables Gradle Plugin Injection by setting a flag to be read via init script.
```
Comment (c00ler) [inline src/main/java/hudson/plugins/gradle/injection/GitScmListener.java:125]:
```suggestion
     * Action that disables Gradle Plugin injection by setting a flag to be read by the init script.
```

File: src/main/java/hudson/plugins/gradle/injection/GitScmListener.java [hunk 2]
```diff
@@ -116,26 +122,37 @@ private static boolean shouldDisableMavenInjection(InjectionConfig config) {
...
+        @Override
+        public void buildEnvironment(@Nonnull Run<?, ?> run, @Nonnull EnvVars envVars) {
+            envVars.put(JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_GRADLE_INJECTION_ENABLED, "false");
         }
 
     }
 
     /**
-     * Marker action to ensure we disable injection in {@link GradleInjectionEnvironmentContributor}.
+     * Action that disables Maven Extension Injection by modifying corresponding environment variables.
```
Comment (c00ler) [inline src/main/java/hudson/plugins/gradle/injection/GitScmListener.java:142]:
```suggestion
     * Action that disables Maven Extension injection by modifying corresponding environment variables.
```

Comment (c00ler) [inline src/main/java/hudson/plugins/gradle/injection/GitScmListener.java:142]:
```suggestion
     * Action that disables Maven Extension injection by modifying corresponding environment variables.
```

File: src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy [hunk 3]
```diff
@@ -583,6 +588,39 @@ node {
...
+        given:
+        mavenInstallationRule.mavenVersion = '3.9.2'
+        mavenInstallationRule.addInstallation()
+        withInjectionConfig {
+            vcsRepositoryFilter = filter
+        }
+        createSlaveAndTurnOnInjection()
+
+        def p = j.createFreeStyleProject()
+        p.buildWrappersList.add(new BuildScanBuildWrapper())
```
Comment (c00ler) [inline src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy:601]: This wrapper is not needed for the test that we are doing. Might only be needed for UI tests.

File: src/main/java/hudson/plugins/gradle/injection/GitScmListener.java (hunk 1 above)
Comment (welandaz) [reply src/main/java/hudson/plugins/gradle/injection/GitScmListener.java]: Done

File: src/main/java/hudson/plugins/gradle/injection/GitScmListener.java (hunk 2 above)
Comment (welandaz) [reply src/main/java/hudson/plugins/gradle/injection/GitScmListener.java]: Done

File: src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy (hunk 3 above)
Comment (welandaz) [reply src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy]: Yup, removed

File: src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy [hunk 4]
```diff
@@ -583,6 +588,39 @@ node {
...
+
+        def slave = createSlave('foo')
+        p.setAssignedNode(slave)
+
+        when:
+        def build = j.buildAndAssertSuccess(p)
+
+        then:
+        if (shouldApplyAutoInjection) {
+            j.assertLogContains("Publishing a build scan to scans.gradle.com requires accepting the Gradle Terms of Service", build)
```
[the remark this answers is not in the document]
Comment (welandaz) [reply src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy]: This is the only place where it's used, so I just moved it to the top of the test as a variable

**Review by c00ler — APPROVED** [review]

**Pull request merged**